journald/lib_journald.cpp

```diff
@@ -163,10 +163,16 @@ class JournaldContainerLoggerProcess :
 
     // NOTE: This field is required by the master/agent, but the protobuf
     // is optional for backwards compatibility.
-    CHECK(executorInfo.has_framework_id());
-    label.set_key("FRAMEWORK_ID");
-    label.set_value(executorInfo.framework_id().value());
-    labels.add_labels()->CopyFrom(label);
+    //
+    // NOTE: It is possible for the ExecutorInfo object to be blank if
+    // a nested container is launched after restarting the Mesos agent.
+    // This is because the agent does not need to keep the ExecutorInfo
+    // checkpointed after it has already launched the executor.
```

Comment:
Is there any race here, or will this reliably happen any time we launch a nested container on a container from the agent's previous start? E.g. restart an agent then try to debug into any UCR container?